apps/aeutils/src/aeu_http_client.erl

```diff
@@ -0,0 +1,72 @@
+-module(aeu_http_client).
+
+%% API
+-export([request/4]).
+
+
+request(BaseUri, Method, Endpoint, Params) ->
+    Timeout = aeu_env:user_config_or_env(
+                [<<"http">>, <<"external">>,<<"request_timeout">>],
+                aehttp, http_request_timeout, 1000),
+    CTimeout = aeu_env:user_config_or_env(
+                 [<<"http">>, <<"external">>, <<"connect_timeout">>],
+                 aehttp, http_connect_timeout, min(Timeout, 1000)),
+    HTTPOptions = [{timeout, Timeout}, {connect_timeout, CTimeout}],
+    request(BaseUri, Method, Endpoint, Params, [], HTTPOptions, []).
+
+request(BaseUri, get, Endpoint, Params, Header, HTTPOptions, Options) ->
+    URL = binary_to_list(
+            iolist_to_binary(
+              [BaseUri, Endpoint, encode_get_params(Params)])),
+    lager:debug("GET URL = ~p", [URL]),
+    R = httpc:request(get, {URL, Header}, HTTPOptions, Options),
+    process_http_return(R);
+request(BaseUri, post, Endpoint, Params, Header, HTTPOptions, Options) ->
+    URL = BaseUri ++ Endpoint,
+    {Type, Body} = case Params of
+                       Map when is_map(Map) ->
+                           %% JSON-encoded
+                           lager:debug("JSON-encoding Params: ~p",[Params]),
+                           {"application/json", jsx:encode(Params)};
+                       [] ->
+                           {"application/x-www-form-urlencoded",
+                            http_uri:encode(Endpoint)}  %% is this correct??
```

Comment:
Please prefer splitting refactoring and code change in separate commits, especially when performing changes difficult to track in git (e.g. portion of file split in other file). Though not needed for this PR (nice to have).